File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2022-08-15/examples-java/CosmosDBTableList.java
import com.azure.core.util.Context;

/** Samples for TableResources ListTables. */
public final class Main {
    /*
     * x-ms-original-file: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2022-08-15/examples/CosmosDBTableList.json
     */
    /**
     * Sample code: CosmosDBTableList.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void cosmosDBTableList(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .cosmosDBAccounts()
            .manager()
            .serviceClient()
            .getTableResources()
            .listTables("rgName", "ddb1", Context.NONE);
    }
}

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-05-15-preview/examples-js/CosmosDBTableList.js
const { CosmosDBManagementClient } = require("@azure/arm-cosmosdb");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Lists the Tables under an existing Azure Cosmos DB database account.
 *
 * @summary Lists the Tables under an existing Azure Cosmos DB database account.
 * x-ms-original-file: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-05-15-preview/examples/CosmosDBTableList.json
 */
async function cosmosDbTableList() {
  const subscriptionId = "subid";
  const resourceGroupName = "rgName";
  const accountName = "ddb1";
  const credential = new DefaultAzureCredential();
  const client = new CosmosDBManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.tableResources.listTables(resourceGroupName, accountName)) {
    resArray.push(item);
  }
  console.log(resArray);
}

cosmosDbTableList().catch(console.error);
